modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CreateTableCommand.java

```diff
@@ -142,6 +144,8 @@ public List<UpdateEntry> get(Catalog catalog) {
                 storageProfile
         );
 
+        updateContext.registerTableCreation(tableId);
```

Comment:
personally, I don't like the idea of maintaining a particular in one command to use in another. It must be done automatically outside of command processing.

The simplest way to address the issue is to keep two different versions of catalog: base one and current one. Base version should be used in CreateIndexCommand to make sure table wasn't presented before, while current is used to apply update entries.

Another option is to analyze UpdateEntries produced